bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/gems/StrehlFeature.java

```diff
@@ -407,30 +408,41 @@ private BandsList getBandpass(GuideProbe.Type type) {
+    // throws an exception if a star's coordinates are unknown
     private Star[] targetListToStarList(List<SPTarget> targetList) {
         Star[] starList = new Star[3];
         int n = targetList.size();
         for (int i = 0; i < starList.length; i++) {
-            starList[i] = i < n ? targetToStar(targetList.get(i)) : null;
+            if (i < n) {
+                final SPTarget spt = targetList.get(i);
+                final Option<Star> op = targetToStar(spt);
+                if (op.isEmpty()) {
+                    throw new RuntimeException("No coordinates for " + spt.getTarget().getName());
+                }
+                starList[i] = op.getValue();
+            }
         }
         return starList;
     }
 
-    // Returns a mascot Star object for the given target.
-    private Star targetToStar(SPTarget target) {
-        String name = target.getTarget().getName();
-        double ra = target.getTarget().getRaDegrees();
-        double dec = target.getTarget().getDecDegrees();
-        double baseX = _tii.getBasePos().getRaDeg();
-        double baseY = _tii.getBasePos().getDecDeg();
-        Magnitude undef = new Magnitude(Magnitude.Band.J, MascotConf.invalidMag());
-        double bmag = target.getTarget().getMagnitude(Magnitude.Band.B).getOrElse(undef).getBrightness();
-        double vmag = target.getTarget().getMagnitude(Magnitude.Band.V).getOrElse(undef).getBrightness();
-        double rmag = target.getTarget().getMagnitude(Magnitude.Band.R).getOrElse(undef).getBrightness();
-        double jmag = target.getTarget().getMagnitude(Magnitude.Band.J).getOrElse(undef).getBrightness();
-        double hmag = target.getTarget().getMagnitude(Magnitude.Band.H).getOrElse(undef).getBrightness();
-        double kmag = target.getTarget().getMagnitude(Magnitude.Band.K).getOrElse(undef).getBrightness();
-        return Star.makeStar(name, baseX, baseY, bmag, vmag, rmag, jmag, hmag, kmag, ra, dec);
+    // Returns a mascot Star object for the given target, if coordinates are known
+    private Option<Star> targetToStar(SPTarget target) {
+        final Option<Long> when = _iw.getContext().schedulingBlockJava().map(SchedulingBlock::start);
+        return
+            target.getTarget().getRaDegrees(when).flatMap(ra ->
+            target.getTarget().getDecDegrees(when).map(dec -> {
+                String name = target.getTarget().getName();
+                double baseX = _tii.getBasePos().getRaDeg();
+                double baseY = _tii.getBasePos().getDecDeg();
+                Magnitude undef = new Magnitude(Magnitude.Band.J, MascotConf.invalidMag());
+                double bmag = target.getTarget().getMagnitude(Magnitude.Band.B).getOrElse(undef).getBrightness();
+                double vmag = target.getTarget().getMagnitude(Magnitude.Band.V).getOrElse(undef).getBrightness();
+                double rmag = target.getTarget().getMagnitude(Magnitude.Band.R).getOrElse(undef).getBrightness();
```

Comment:
This is unrelated to this particular PR but I think this should follow the "use one of the R-bands" logic